images/daemon/pkg/daemon/kata_actions.go

```diff
@@ -0,0 +1,71 @@
+package daemon
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	kataTypes "github.com/openshift/kata-operator/api/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// KataActions declares the possible actions the daemon can take.
+type KataActions interface {
+	Install(kataConfigResourceName string) error
+	Upgrade() error
+	Uninstall(kataConfigResourceName string) error
+}
+
+type updateStatus = func(a *kataTypes.KataConfigStatus)
+
+func updateKataConfigStatus(kataClient client.Client, kataConfigResourceName string, us updateStatus) (err error) {
+	var kataConfig kataTypes.KataConfig
+	attempts := 5
+	for i := 0; i < attempts; i++ {
+		err = kataClient.Get(context.Background(), client.ObjectKey{
+			Name: kataConfigResourceName,
+		}, &kataConfig)
+
+		if err != nil {
+			continue // Error, let's attempt to get the object again. TODO - maybe check error type and decide?
+		}
+
+		us(&kataConfig.Status)
+
+		err = kataClient.Status().Update(context.Background(), &kataConfig)
+
+		if err == nil {
+			break
+		}
+
+		time.Sleep(5 * time.Second)
+	}
+
+	return err
+}
```

Comment:
I see no downside to your suggestion. I think  5 attempts was just chosen as a pragmatic number to start with, same for the 5 seconds. Any objection to this suggestion @harche ?

Reply:
I'd probably make this change as a follow-on PR and not as part of this huge copy/paste PR. Then we could better test it. What do you think?

Reply:
Sure, was just a suggestion :)

Reply:
@jensfr you can decide what's the best timeout and number of attempts values should be. For 3 worker nodes, we have 3 daemons that are trying to update the same field. Because of that sometimes one of the daemons end up working on the outdated version of values. 

But I like @zanetworker's suggestion to use `wait.Poll` here.

Reply:
Then I'll leave it as is now and figure out all of this as a separate work item. Thanks for the feedback @harche !